### ISSUE TITLE
local -n is not handled properly (bash)

### ISSUE THREAD

**Issue by pan64** [koalaman/shellcheck#2712]
#### For bugs
- Rule Id (if any, e.g. SC1000):  SC2034
- My shellcheck version (`shellcheck --version` or "online"): online
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash

function test_func {
        local -n a=$1
        read -r a < <(echo this is something)
}

BB="nothing"
test_func BB
echo "$BB" # this is something

```

#### Here's what shellcheck currently says:

[Line 5:](javascript:setPosition(5, 17))
        read -r a < <(echo this is something)
                ^-- [SC2034](https://www.shellcheck.net/wiki/SC2034) (warning): a appears unused. Verify use (or export if used externally).

#### Here's what I wanted or expected to see:

BB is used, therefore it is a false warning
